go-controller/pkg/ovn/policy.go

```diff
@@ -1262,6 +1262,9 @@ func (oc *Controller) deleteNetworkPolicy(policy *knet.NetworkPolicy, np *networ
 	if ok {
 		expectedLastPolicyNum = 1
 		np = foundNp
+	} else {
+		klog.Warningf("Unable to delete network policy: %s/%s since its not found in cache", policy.Namespace, policy.Name)
```

Comment:
This is clearly NOT the right thing to do, I'm not 100% if I need to insert a value based on the kapi object and pass that into destroyNetworkPolicy to ensure we remove anything stale that might be there that was half way created for the policy.

NOTE: This issue gets triggered because of https://github.com/openshift/ovn-kubernetes/pull/1043/files and not sure why we didn't block edge upgrades for this. We probably don't want users to delete and recreate their NPs manually right?

Reply:
this needs to not be an else, because np could still be legit if passed inot the function. Just check if np ==nil in a separate block

Reply:
[7:37 PM] createDefaultDenyPGAndACLs call needs a defer after it to delete it if create fails
[7:37 PM] 2. deleteNetworkPolicy needs to check if np == nil before it calls destroyNetworkPolicy and if true, then just return nil
3. fix this part:
if np == nil {
			return fmt.Errorf("failed to get namespace lock when deleting policy %s in namespace %s",
				policy.Name, policy.Namespace)
		}